src/csm_testing/tests/check_workflow_template/__main__.py

```diff
@@ -0,0 +1,314 @@
+    """Kills the pod if the main container is finished"""
+    v1 = client.CoreV1Api()  # pylint: disable=invalid-name
+
+    pods = v1.list_namespaced_pod(
+        namespace=namespace,
+        label_selector=f"workflows.argoproj.io/workflow={workflow_name}",
+    ).items
+
+    for pod in pods:
+        pod_name = pod.metadata.name
+        print(f"INFO: Checking pod: {pod_name}")
+
+        # Check if the main container has finished
+        if is_main_container_finished(pod_name, namespace):
+            print(f"INFO: Main container finished in pod {pod_name}. Deleting pod.")
+            v1.delete_namespaced_pod(name=pod_name, namespace=namespace)
+            print(f"INFO: Pod {pod_name} deleted.")
+            return True
+        print(f"INFO: Main container still running in pod {pod_name}.")
+    return False
+
+
+def update_image_version_in_template(
+    workflow_template_str, old_image_prefix, old_version, new_version
+):
+    """
+    Find the image in the workflowtemplate and replace its version
+    with the user provided version
+    """
+    updated_workflow_template_str = workflow_template_str.replace(
+        old_image_prefix + old_version, old_image_prefix + new_version
+    )
+    return updated_workflow_template_str
+
+
+def wait_for_workflow_to_succeed(namespace, workflow_name, timeout=6000, interval=20):
+    """
+    Polls the workflow status until it succeeds or the timeout is reached.
+    """
+    start_time = time.time()
+    while time.time() - start_time < timeout:
+        try:
+            # Get the workflow status
+            command = [
+                "kubectl",
+                "get",
+                "workflow",
+                workflow_name,
+                "-n",
+                namespace,
+                "-o",
+                "jsonpath={.status.phase}",
+            ]
+            result = subprocess.run(
+                command,
+                check=True,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                universal_newlines=True,
+            )
+            status = result.stdout.strip()
+
+            # Check the status of the workflow
+            print(f"Workflow {workflow_name} status: {status}")
+
+            if status == "Succeeded":
+                print(f"INFO: Workflow {workflow_name} succeeded.")
+                return True
+
+            if status in ["Failed", "Error"]:
+                print(f"ERROR: Workflow {workflow_name} failed with status {status}.")
+                return False
+
+            if status == "Running":
+                print("INFO: Check if main is done")
+                if check_and_kill_pod(workflow_name, namespace):
+                    print(
+                        f"WARNING: {workflow_name} is running but the main container is done."
+                    )
+                    return True
+        except ApiException as err:
+            print(f"ERROR: Exception when fetching workflow status: {err}")
```

Comment:
[This comment from a previous review](https://github.com/Cray-HPE/csm-testing/pull/629#discussion_r1873822841) was not addressed.

Reply:
I have modified the code to wrap the kubectl command in try-Exception block.